gunicorn/arbiter.py

```diff
@@ -378,7 +380,11 @@ def stop(self, graceful=True):
         killed gracefully  (ie. trying to wait for the current connection)
         """
 
-        unlink = self.reexec_pid == self.master_pid == 0 and not self.systemd
```

Comment:
This trigger the question of the supervision and USR2 signal. Right now (except in the case of fds given via systemd or others) we are making sure Gunicorn is the owner of a port and it can't be stolen  by another process. But if the master doesn't listen first what will happen in USR2 if all workers of the old master have been killed? We still want to be able to go back on the old master and own the port or be the first to listen. Maybe we can just also listen on the socket in the master to just supervise the socket? Related quest can't we just rebind the socket / process and start to listen on the master?